bundles/org.eclipse.cdt.lsp.clangd/src/org/eclipse/cdt/lsp/clangd/internal/config/ClangFormatFileHandler.java

```diff
@@ -25,21 +25,58 @@
 import org.eclipse.core.runtime.NullProgressMonitor;
 import org.eclipse.core.runtime.Platform;
 import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.IJobChangeEvent;
+import org.eclipse.core.runtime.jobs.JobChangeAdapter;
+import org.eclipse.lsp4e.LSPEclipseUtils;
+import org.eclipse.swt.widgets.Display;
+import org.osgi.service.component.annotations.Component;
 
-public final class ClangFormatUtils {
+@Component(property = { "service.ranking:Integer=0" })
+public final class ClangFormatFileHandler implements ClangFormatFile {
 	public static final String format_file = ".clang-format"; //$NON-NLS-1$
 
+	private class OpenFileExecuter extends JobChangeAdapter {
+		private final IFile file;
+
+		public OpenFileExecuter(IFile file) {
+			this.file = file;
+		}
+
+		@Override
+		public void done(IJobChangeEvent event) {
+			if (Status.OK_STATUS.equals(event.getResult())) {
+				Display.getDefault().asyncExec(new Runnable() {
+					@Override
+					public void run() {
+						LSPEclipseUtils.open(file.getLocationURI().toString(), null);
+					}
+				});
+			}
+		}
+
+	}
+
 	/**
-	 * Checks if the formatFile exists. Creates it if not.
+	 * Opens the .clang-format file in the given project. Creates a file with default values, if not yet existing prior to the opening.
 	 * @param formatFile
 	 */
-	public Optional<IFile> getClangFormatFile(IProject project) {
-		var formatFile = project.getFile(format_file);
-		var status = createFileFromResource(formatFile);
-		return status.isOK() ? Optional.of(formatFile) : Optional.empty();
+	@Override
+	public void openClangFormatFile(IProject project) {
+		var job = getClangFormatFileCreatorJob(project);
+		job.addJobChangeListener(new OpenFileExecuter(project.getFile(format_file)));
```

Comment:
`WorkspaceJob` already has `performDone` method to be called on done

Reply:
But it can't be overriden because its final.

Reply:
Sorry, I cannot find a `performDone` method in `WorkspaceJob` nor in its parent classes  which can be overridden.

Reply:
Sorry, this is my fault, I tried to mention `WorkbenchJob` https://github.com/eclipse-platform/eclipse.platform.ui/blob/3a6db4376cd1675afc592adf472c83c46d6c57a4/bundles/org.eclipse.ui.workbench/Eclipse%20UI/org/eclipse/ui/progress/WorkbenchJob.java#L83